src/main/mondrian/rolap/SqlTupleReader.java

```diff
@@ -90,6 +90,16 @@
     private int missedMemberCount;
     private static final String UNION = "union";
     private int emptySets = 0;
+    // allow hints by default
+    private boolean allowHints = true;
+
+    public boolean isAllowHints() {
+        return allowHints;
+    }
+
+    public void setAllowHints(boolean allowHints) {
+        this.allowHints = allowHints;
+    }
 
```

Comment:
@akhayrutdinov 
It looks like the property and getter/setter are all new with this PR, but I'm not seeing anywhere they are used.  Not sure what you intended with them.